netket/_vmc.py

```diff
@@ -211,11 +243,14 @@ def iter(self, n_steps, step=1):
             self.advance(step)
             yield self.step_count
 
-    def add_observable(self, name, obs):
+    def add_observable(self, obs, name):
         """
         Add an observables to the set of observables that will be computed by default
         in get_obervable_stats.
         """
+        if type(name) != str:
```

Comment:
This ad-hoc type-checking is not very Pythonic. In any case, if we have to check the type, a better way would be to write `if isinstance(name, str): ...`.

Reply:
Good point, the thing is whether we should use type annotations or instead do something else? Maybe just a runtime check of the input value?

Reply:
Alright actually I agree that the dict should not be restricted to deal with str. I will remove this

Reply:
> Alright actually I agree that the dict should not be restricted to deal with str. I will remove this
Yes, I think not checking the key okay for now.